frontend/src/components/CheckboxGroup/CheckboxGroup.tsx

```diff
@@ -38,14 +41,20 @@ const StyledSegment = styled(Segment)`
 `;
 
 const CheckboxGroup: FunctionComponent<ComponentProps> = (props) => {
+  const [checkboxOptions, setCheckboxOptions] = useState<CheckboxGroupOption[]>(props.options);
   const [checkboxes, addCheckboxes] = useState<string[] | undefined>(
     props.selectedOptions && props.selectedOptions.length > 0 ? props.selectedOptions : [],
   );
+  const [selectAll, setSelectAll] = useState<boolean>(false);
 
   useEffect(() => {
     addCheckboxes(props.selectedOptions);
```

Comment:
In here, add a check for the length of `props.selectedOptions` vs `props.options` and use this to set the value of `selectAll` i.e.
`setSelectAll(props.selectedOptions.length === props.options.length)` but only if the props.options length isn't 0

Reply:
Added set state in useEffect